File: app/src/main/java/app/opass/ccip/model/Speaker.java

package app.opass.ccip.model;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

public class Speaker {

    @SerializedName("id")
    @Expose
    private String id;
    @SerializedName("avatar")
    @Expose
    private String avatar;
    @SerializedName("zh")
    @Expose
    private Zh_ zh;
    @SerializedName("en")
    @Expose
    private En_ en;

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getAvatar() {
        return avatar;
    }

    public void setAvatar(String avatar) {
        this.avatar = avatar;
    }

    public SpekaerDetail getZh() {
        return zh;
    }

    public void setZh(Zh_ zh) {
        this.zh = zh;
    }

    public SpekaerDetail getEn() {
        return en;
    }

    public void setEn(En_ en) {
        this.en = en;
    }

}

File: app/src/main/java/app/opass/ccip/model/Speaker.kt
package app.opass.ccip.model

import android.content.Context
import app.opass.ccip.util.LocaleUtil
import java.util.Locale

data class Speaker(
    val id: String,
    val avatar: String,
    val zh: Zh_,
    val en: En_
) {
    fun getSpeakerDetail(context: Context): SpeakerDetail {
        return when (LocaleUtil.getCurrentLocale(context).language) {
            "nan", "zh" -> zh
            else -> en
        }
    }
}
